src/api/routes/tx.ts

```diff
@@ -238,19 +238,28 @@ export function createTxRouter(db: DataStore): RouterWithAsync {
     const eventLimit = parseTxQueryEventsLimit(req.query['event_limit'] ?? 96);
     const eventOffset = parsePagingQueryInput(req.query['event_offset'] ?? 0);
     const includeUnanchored = isUnanchoredRequest(req, res, next);
-    const txQuery = await searchTx(db, { txId: tx_id, eventLimit, eventOffset, includeUnanchored });
-    if (!txQuery.found) {
-      res.status(404).json({ error: `could not find transaction by ID ${tx_id}` });
-      return;
-    }
-    // TODO: this validation needs fixed now that the mempool-tx and mined-tx types no longer overlap
-    /*
+    try {
+      const txQuery = await searchTx(db, {
+        txId: tx_id,
+        eventLimit,
+        eventOffset,
+        includeUnanchored,
+      });
+      if (!txQuery.found) {
+        res.status(404).json({ error: `could not find transaction by ID ${tx_id}` });
+        return;
+      }
+      // TODO: this validation needs fixed now that the mempool-tx and mined-tx types no longer overlap
+      /*
     const schemaPath = require.resolve(
       '@stacks/stacks-blockchain-api-types/entities/transactions/transaction.schema.json'
     );
     await validate(schemaPath, txQuery.result);
     */
-    res.json(txQuery.result);
+      return res.json(txQuery.result);
+    } catch (error: any) {
+      res.status(400).json(error.message);
```

Comment:
The main problem I see here is blindly catching all errors and returning a 400. The `searchTx` function should return a `found: false` value for 4xx errors. If it throws then it's likely a bug which should be a 5xx internal server error. Or it's because of bad user input which should be explicitly checked beforehand in order to return a 400. 

For example, the query param parsing above this `try` statement is where the Error is actually thrown, which results in the problem described in the issue https://github.com/hirosystems/stacks-blockchain-api/issues/851.

Wrapping all these routes in a `try ... catch { return 400 }` is the wrong approach. Seems like there's two reasonable approaches: 

A) Refactor the functions that parse the query params so that they return something indicating bad input rather than throwing, and modify the route handlers to check for the bad input results and return a 400.

B) Create a custom Error instance that can indicate to the Express error handler that the error already triggered a 400 response, and avoid logging an `level: error` message.

Reply:
@zone117x I have worked with your second suggested approach. Can you just go through the code once and let me know if this looks good? If you're fine with it, I will use this `RespError` class for throwing errors in my other PR, too. 

I have also added status code and `RespErrorType` into the class obj, to use different status codes for any such issue in the future that requires a different status code or treatment.

Reply:
Left a comment, generally looks like you're headed in the right direction.